### DISCUSSION TITLE
[BUG] Set-PnPListItemPermission fails when AAD user is disabled

### DISCUSSION THREAD

**Discussion by jeremyhagan** [pnp/powershell#2400] [General]
### Reporting an Issue or Missing Feature
When using Set-PnPListItemPermission to add permissions to a document library folder for a licensed AAD user that is sign in disabled (or very recently enabled) the cmdlet fails and throw an error "The specified user Test.User@example.com could not be found"

Please note that the user must not be already in the list as shown by Get-PnPUser for this failure to occur.

### Expected behavior 
The user is a licensed SharePoint user and should be added to the permissions for the list item.

### Actual behavior 
Error is thrown

### Steps to reproduce behavior
Create and license a new AAD user with SharePoint plan 2 and leave it as sign in disabled. Wait 15 minutes.
Add permissions
Set-PnPListItemPermission -List $DocLibrary -Identity $FolderId -User test.user@example.com -AddRole "Read" -Connection $SPConnection 
Error is thrown
Enable the user in ADD and immediately rerun the command
Error is thrown
Wait 5 minutes then rerun the command
Command is successful
Disable the user in AAD
Remove the user from the site (Remove-PnPUser)
Wait 5 minutes
Rerun the command
Error is thrown

### What is the version of the Cmdlet module you are running?
1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [X ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by fowl2** [comment]
I'm afraid that this is the SharePoint behavior. Can you replicate it in the user interface?

**Comment by gautamdsheth** [comment]
@jeremyhagan  - Its not a bug but standard SharePoint behavior , moving this to discussion forum.
